### PR TITLE
Add new themes

### PR DESCRIPTION
**Pull request by Shluffy**
### Resolves:

add some new themes from theme discussion

### Changes:

add the scratch and terminal theme

### Local Tests:
Known Bugs: 
~~Semicolons should be after the css variables.~~
~~Terminal is spelled terrminal~~

tested ~~but bugs found.~~

### CODE REVIEW

**Review by leahcimto — CHANGES_REQUESTED** [review]
Upon previewing locally, the themes were not displayed correctly. This is because you are missing semicolons after each variable in the stylesheet.

**Comment by gosoccerboy5** [comment]
> Semicolons should be added after each css variable.

That's what happens when you rely too much on [ASI](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#automatic_semicolon_insertion) in JS :P

**Comment by Shluffy** [comment]
@leahcimto fixed. I don't know how to dismiss it so i re-requested your review. I am going to wait for the github pages to update before marking this ready for review

**Comment by BarelySmooth** [comment]
The Scratch Theme looks really cool! Its my new favourite theme 🎉

I thing I noticed was that the button had the same color as the footer in the Scratch Theme and the terminal theme. I think you should change it, You can modify ```--tertiary``` and ```--tertiaryHover``` CSS variables to fix this. See [this](https://github.com/theforumhelpers/theforumhelpers.github.io/wiki/Theme-Guide) wiki page for a description of all the variables used.
image

For the terminal theme, the contrast is a little too low for the nav bar text and the footer text. A contrast of 1.66 is a little too low in my opinion.
image

**Comment by Shluffy** [comment]
@BarelySmooth i just noticed that. I didn't make these themes. I am just implementing them. 
@leahcimto can you look for what you want for the scratch theme
@Steve0Greatness can you pick for terminal theme

**Comment by Steve0Greatness** [comment]
> @BarelySmooth i just noticed that. I didn't make these themes. I am just implementing them.
> @leahcimto can you look for what you want for the scratch theme
> @Steve0Greatness can you pick for terminal theme

something like:
Screenshot 2021-06-14 10 02 13 AM
maybe?

**Comment by Shluffy** [comment]
@Steve0Greatness i think it looks good. Please post the code here so I can update this

**Comment by Accio1** [comment]
> > @BarelySmooth i just noticed that. I didn't make these themes. I am just implementing them.
> > @leahcimto can you look for what you want for the scratch theme
> > @Steve0Greatness can you pick for terminal theme
> 
> something like:
> Screenshot 2021-06-14 10 02 13 AM
> maybe?

Isn't the whole point of a terminal theme that the text is green?

**Comment by Steve0Greatness** [comment]
> > > @BarelySmooth i just noticed that. I didn't make these themes. I am just implementing them.
> > > @leahcimto can you look for what you want for the scratch theme
> > > @Steve0Greatness can you pick for terminal theme
> > 
> > 
> > something like:
> > maybe?
> 
> Isn't the whole point of a terminal theme that the text is green? You could just use green for both the primary and secondary text colours.

not 100% of the time, but here's the here's the page with the primary and secondary text colors rearranged:
Screenshot 2021-06-14 11 11 20 AM

File: resources/stylesheet.css
Comment (gosoccerboy5) [inline resources/stylesheet.css:35]: I think a color closer to #7af850 would be a bit better. green is a little dark for my preference

**Comment by Steve0Greatness** [comment]
@Shluffy here's the css:
```css
html.terrminal {
    --primary: #121212;
    --secondary: #333;
    --secondaryHover: #555;
    --textPrimary: #90f;
    --textSecondary: #32cd32;
    --linkHover: #a1c6ff;
    --searchBar: #222b2e;
    --saturation: 60%;
    --tertiary: #2a2b2d;
    --tertiaryHover: #565a60;
}
```
[preview](https://user-images.githubusercontent.com/75220768/121930817-a90b2380-ccf7-11eb-9af4-201f5fd6a626.png)
@Accio1 the one you asked for:
```css
html.terrminal {
    --primary: #121212;
    --secondary: #333;
    --secondaryHover: #555;
    --textPrimary: #32cd32;
    --textSecondary: #90f;
    --linkHover: #a1c6ff;
    --searchBar: #222b2e;
    --saturation: 60%;
    --tertiary: #2a2b2d;
    --tertiaryHover: #565a60;
}
```
[preview](https://user-images.githubusercontent.com/75220768/121939099-749c6500-cd01-11eb-8ce0-06d1b8ef0f39.png)
feel free to use the one you like the most.

File: resources/stylesheet.css
Comment (Shluffy) [reply resources/stylesheet.css:35]: @gosoccerboy5 see [this](https://github.com/theforumhelpers/theforumhelpers.github.io/pull/243#issuecomment-860894615). if it is still relevant ask @Steve0Greatness

**Comment by gosoccerboy5** [comment]
@Steve0Greatness it's spelled "terminal" not "terrminal", just letting you know before this snowballs into distaster

File: resources/stylesheet.css
Comment (Shluffy) [reply resources/stylesheet.css:35]: @gosoccerboy5 `#32cd32` is what the green is going to look like. Do you still think it should be changed

**Comment by Shluffy** [comment]
> @Steve0Greatness it's spelled "terminal" not "terrminal", just letting you know before this snowballs into distaster

i did at first typo that, but I will change it when i enter it. Also you should add `Mode` to the end for the css to load

**Comment by Shluffy** [comment]
@Steve0Greatness i think the purple is too bright, so I think green would be good

**Comment by Steve0Greatness** [comment]
> @Steve0Greatness i think the purple is too bright, so I think green would be good

sounds good.

**Comment by Shluffy** [comment]
anyone have any ideas for the last three for scratch mode? scratch doesn't have a change theme button

**Comment by gosoccerboy5** [comment]
all right this looks great

**Comment by Shluffy** [comment]
@leahcimto color for the buttons for scratch mode?

**Comment by BarelySmooth** [comment]
> anyone have any ideas for the last three for scratch mode? scratch doesn't have a change theme button

```saturation``` should stay at hundred.
```tertiary``` can be #4280D9
```tertiaryhover``` can be #2d65b3

Or you should just wait until leahcimto responds :P

image

Terminal looks good!

File: resources/stylesheet.css
Comment (gosoccerboy5) [reply resources/stylesheet.css]: > @gosoccerboy5 `#32cd32` is what the green is going to look like. Do you still think it should be changed

oh wow I didn't know you can preview colours like that

**Comment by gosoccerboy5** [comment]
is it possible to make the code a bit more consistent while you're at it?
https://github.com/theforumhelpers/theforumhelpers.github.io/blob/d1b448f013809fa6bc418e0c8bb96d62a49290c5/resources/theme.js#L13
vs
https://github.com/theforumhelpers/theforumhelpers.github.io/blob/d1b448f013809fa6bc418e0c8bb96d62a49290c5/resources/theme.js#L18
 \- one has a semicolon, one doesn't.
And can we fix this lol:
https://github.com/theforumhelpers/theforumhelpers.github.io/blob/d1b448f013809fa6bc418e0c8bb96d62a49290c5/resources/theme.js#L38-L40

**Comment by Accio1** [comment]
> is it possible to make the code a bit more consistent while you're at it?
> https://github.com/theforumhelpers/theforumhelpers.github.io/blob/d1b448f013809fa6bc418e0c8bb96d62a49290c5/resources/theme.js#L13
> 
> 
> vs
> https://github.com/theforumhelpers/theforumhelpers.github.io/blob/d1b448f013809fa6bc418e0c8bb96d62a49290c5/resources/theme.js#L18
> 
> 
> - one has a semicolon, one doesn't.
> And can we fix this lol:
> https://github.com/theforumhelpers/theforumhelpers.github.io/blob/d1b448f013809fa6bc418e0c8bb96d62a49290c5/resources/theme.js#L38-L40

This would be best addressed in a separate issue/PR.

File: resources/stylesheet.css
Comment (BarelySmooth) [inline resources/stylesheet.css:54]: You should use an uppercase H for the variable :P

```suggestion
    --tertiaryHover: #2d65b3;
```

**Comment by leahcimto** [comment]
> > anyone have any ideas for the last three for scratch mode? scratch doesn't have a change theme button
> 
> `saturation` should stay at hundred.
> `tertiary` can be #4280D9
> `tertiaryhover` can be #2d65b3
> 
> Or you should just wait until leahcimto responds :P
> 
> image
> 
> Terminal looks good!

@BarelySmooth @Shluffy I'm ok with this. Sorry for the late reply, I've been busy.

**Comment by BarelySmooth** [comment]
@Shluffy I don't think this should be a draft any longer. It looks complete!

**Review by leahcimto — CHANGES_REQUESTED** [review]
Almost everything looks good!

One thing though, could the text on here be white? It is pretty hard to read when it is that gray color.
Screen Shot 2021-06-18 at 9 23 22 AM

**Comment by CodeGuy92** [comment]
wait how do i test that locally

**Comment by CodeGuy92** [comment]
In the Terminal Theme, when you hover on a link, it turns cyan instead of light green. That should also be fixed.

**Comment by gosoccerboy5** [comment]
> In the Terminal Theme, when you hover on a link, it turns cyan instead of light green. That should also be fixed.

The CSS shows that it is intentional (I think)

**Comment by leahcimto** [comment]
Also, just a heads up to everyone using a browser extension to pick colors, many of those have malware. I used one and it had malware so I recommend using inspect to be safe, or use digital color meter on macOS 11.4 and higher.

**Comment by gosoccerboy5** [comment]
Firefox uses a builtin color picker: hamburger menu at top -> more tools -> eyedropper

**Comment by Steve0Greatness** [comment]
@leahcimto I already use inspect.

**Comment by gosoccerboy5** [comment]
when does this merge?

**Comment by Steve0Greatness** [comment]
> when does this merge?

I think all the reviewers need to say yes to it, especially the code owner.

**Comment by leahcimto** [comment]
> when does this merge?

Requested changes have to be resolved.

**Comment by BarelySmooth** [comment]
> Also, just a heads up to everyone using a browser extension to pick colors, many of those have malware. I used one and it had malware so I recommend using inspect to be safe, or use digital color meter on macOS 11.4 and higher.

Also, on Windows [Microsoft Powertoys Color Picker tool](https://docs.microsoft.com/en-us/windows/powertoys/color-picker) would be the best solution. (Its requires [a download](https://docs.microsoft.com/en-us/windows/powertoys/install) though and can only be used on Windows 10)

**Comment by gosoccerboy5** [comment]
https://github.com/theforumhelpers/theforumhelpers.github.io/blob/e849ebda3cbc306e8cf3b7bdccd5782da5954c4f/resources/stylesheet.css#L18-L53
Again, should this be addressed in a new issue? Anyways, this indentation is inconsistent and rather hurts my eyes. I guess you could get away with it because it's CSS and doesn't require multiple indentation levels, but still, I feel like it should be fixed eventually
 - mixing tabs/spaces
 - the second rule has 4 space indentation
 - the third rule has no indentation(!)

**Comment by Shluffy** [comment]
@Accio1 can this pls be reviewed?

**Comment by Accio1** [comment]
> @Accio1 can this pls be reviewed?

I wasn't aware that you wanted me to review it yet. I'll do it now.

**Comment by Shluffy** [comment]
> > @Accio1 can this pls be reviewed?
> 
> I wasn't aware that you wanted me to review it yet. I'll do it now.

that was before the indentation issue. I can add it in this or another pr

**Comment by Accio1** [comment]
@Shluffy you have to edit this PR to click the box that says "Allow edits from maintainers" so that I can push changes.

**Comment by Shluffy** [comment]
@Accio1 done

**Comment by Accio1** [comment]
I just made a few small changes in my most recent commit (I'm not sure why it is showing up as 2 hours ago because I just pushed it). If no one has any other changes that they would like made, then this is good to merge.

@Shluffy @leahcimto @Steve0Greatness @gosoccerboy5

**Review by BarelySmooth — APPROVED** [review]
I tested it locally and this looks good! I can't wait to see the scratch theme on the website :P

I think leahcimto's change should now be marked as resolved.

**Comment by Shluffy** [comment]
@Accio1 lgtm

**Review by leahcimto — APPROVED** [review]
Looks good!

**Review by Accio1 — APPROVED** [review]

**Comment by Accio1** [comment]
Thank you everybody for your contributions!

**Comment by leahcimto** [comment]
Woohoo!

**Pull request merged**
